docs/developer/style_guide.md

```diff
@@ -0,0 +1,171 @@
+---
+id: dev_style_guide
+title: Style Guide 
+---
+
+**TL;DR**  - Install the provided pre-commits, follow their recommendations
+
+# General Style 
+## Spacing
+* Include 2 lines between functions outside of classes.
+* Include 1 line between class methods.
+* Include 2 lines between the import block and main code body. 
```

Comment:
I forgot to say why I think we might not want to do this: if we write out the automatic rules in the style guide, then we need to maintain both the automatic checks and the rules in the style guide.

A good compromise could be to add a link in the documentation that points to the linting config, so that contributors can check the current automatic style rules straight from the source